File: app/src/main/java/za/org/grassroot2/services/OfflineReceiver.java
package za.org.grassroot2.services;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import javax.inject.Inject;

import za.org.grassroot2.GrassrootApplication;
import za.org.grassroot2.util.UserPreference;

public class OfflineReceiver extends BroadcastReceiver {

    @Inject UserPreference userPrefs;

    public static final String ACTION_RESET_OFFLINE_INFO = "za.org.grassroot2" + ".RESET_OFFLINE_INFO";

    @Override
    public void onReceive(Context context, Intent intent) {
        ((GrassrootApplication)context.getApplicationContext()).getAppComponent().inject(this);
        userPrefs.setNoConnectionInfoDisplayed(false);
    }
}

File: app/src/main/java/za/org/grassroot2/services/OfflineReceiver.kt
package za.org.grassroot2.services

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import za.org.grassroot2.GrassrootApplication
import za.org.grassroot2.util.UserPreference
import javax.inject.Inject

class OfflineReceiver : BroadcastReceiver() {

    @Inject lateinit var userPrefs: UserPreference

    override fun onReceive(context: Context, intent: Intent) {
        (context.applicationContext as GrassrootApplication).appComponent.inject(this)
        userPrefs.setNoConnectionInfoDisplayed(false)
    }

    companion object {
        const val ACTION_RESET_OFFLINE_INFO = "za.org.grassroot2" + ".RESET_OFFLINE_INFO"
    }
}
